### ISSUE TITLE
Project References - Infinite build cycle/loop, hard to debug

### ISSUE THREAD

**Issue by natew** [microsoft/TypeScript#32400]
**TypeScript Version:**  3.5.3

I can't really replicate it as I can't find what's causing it after some searching. It's in a monorepo of about 100 packages and apps.

I can run with `tsc -b --extendedDiagnostics --traceResolution` and I see a massive output, it takes 10 minutes to build when normally took about 50s. For some reason it does still finish, but it's outputting the same resolutions/errors over and over.

I've run into this before in different forms. Sometimes moving a file in VSCode from one package to another will cause it to update an `includes` to reference another file.

One thing I've noticed too is importing the package you're currently in causes similar types of bugs. I've been looking for both of these and haven't found anything yet.

Basically, Typescript is the right level to handle this. It should error when it detects a cycle and show you where to it happens.

**Expected behavior:**

Don't accept cycles in project references.

**Actual behavior:**

Build goes on for a long time.

**Comment by RyanCavanaugh** [comment]
There's already code to detect cycles in project reference graphs and error when this happens. We'd need some way to reproduce this problem to be able to investigate further
